tsdb/engine/tsm1/wal_test.go

```diff
@@ -701,6 +703,124 @@ func TestWALRollSegment(t *testing.T) {
 	}
 }
 
+func TestWAL_DistSize(t *testing.T) {
+	test := func(w *tsm1.WAL, oldZero, curZero bool) () {
+		// get disk size by reading file
+		files, err := ioutil.ReadDir(w.Path())
+		if err != nil {
+			t.Fatalf("read dir err: %v", err)
+		}
+
+		sort.Slice(files, func(i, j int) bool {
+			return files[i].Name() < files[j].Name()
+		})
+
+		var old, cur int64
+		for i, f := range files {
+			if i == len(files)-1 {
+				cur = f.Size()
+			} else {
+				old += f.Size()
+			}
+		}
+
+		// test zero size condition
+		if oldZero && old > 0 {
+			t.Fatalf("expect old disk size: 0, got: %d", old)
+		}
+		if !oldZero && old == 0 {
+			t.Fatalf("expect old disk size larger than 0, got 0")
+		}
+
+		if curZero && cur > 0 {
+			t.Fatalf("expect current disk size: 0, got: %d", cur)
+		}
+		if !curZero && cur == 0 {
+			t.Fatalf("expect current disk size larger than 0, got 0")
+		}
+
+		// test method DiskSizeBytes
+		if got, exp := w.DiskSizeBytes(), old+cur; got != exp {
+			t.Fatalf("expect total disk size: %d, got: %d", exp, got)
+		}
+
+		// test Statistics
+		ss := w.Statistics(nil)
```

Comment:
What is the `nil` here?

Reply:
`nil` means no need to add additional tags